bpfd/src/server/bpf.rs

```diff
@@ -319,6 +319,10 @@ impl<'a> BpfManager<'a> {
                         BpfdError::Error("unable to delete persisted dispatcher data".to_string())
                     })?;
                 }
+                if !self.dispatchers.contains_key(&if_index) {
```

Comment:
You can include this in the `if let Some(old) = old {}` clause above.
We've already removed the key at &if_index so this `if`condition is essentially always true.

The only reason NOT to include it above would be if we wanted to cleanup pinned files even if we've run the cleanup code before, which I don't think is necessary.

Reply:
@dave-tucker done. Soon as IT merges I will get this tested as well. Ty!